addons/l10n_ec/migrations/3.4/post-migration.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, SUPERUSER_ID
+
+def update_withhold_type(env):
+    # reclassifies withhold taxes into independent tax groups for sales and purchases
+    env.cr.execute('''
+        UPDATE account_tax
+        SET tax_group_id=t.id FROM (SELECT id FROM account_tax_group WHERE l10n_ec_type='withhold_vat_sale') AS t
+        WHERE account_tax.id IN (SELECT id FROM account_tax WHERE tax_group_id IN (SELECT id FROM account_tax_group WHERE l10n_ec_type='withhold_vat') AND type_tax_use='sale')
+    ''')
+    env.cr.execute('''
+        UPDATE account_tax
+        SET tax_group_id=t.id FROM (SELECT id FROM account_tax_group WHERE l10n_ec_type='withhold_vat_purchase') AS t
+        WHERE account_tax.id IN (SELECT id FROM account_tax WHERE tax_group_id IN (SELECT id FROM account_tax_group WHERE l10n_ec_type='withhold_vat') AND type_tax_use='purchase')
+    ''')
+    env.cr.execute('''
+        UPDATE account_tax
+        SET tax_group_id=t.id FROM (SELECT id FROM account_tax_group WHERE l10n_ec_type='withhold_income_sale') AS t
+        WHERE account_tax.id IN (SELECT id FROM account_tax WHERE tax_group_id IN (SELECT id FROM account_tax_group WHERE l10n_ec_type='withhold_income_tax') AND type_tax_use='sale')
+    ''')
+    env.cr.execute('''
+        UPDATE account_tax
+        SET tax_group_id=t.id FROM (SELECT id FROM account_tax_group WHERE l10n_ec_type='withhold_income_purchase') AS t
+        WHERE account_tax.id IN (SELECT id FROM account_tax WHERE tax_group_id IN (SELECT id FROM account_tax_group WHERE l10n_ec_type='withhold_income_tax') AND type_tax_use='purchase')
+    ''')
+
+def update_type_tax_use(env):
+    # sets type_tax_use = none for withholding taxes
+    env.cr.execute('''
+        UPDATE account_tax
+        SET type_tax_use = 'none'
+        WHERE tax_group_id IN (SELECT id FROM account_tax_group WHERE l10n_ec_type IN ('withhold_income_purchase','withhold_vat_purchase','withhold_income_sale','withhold_vat_sale'))
+    ''')
+
+def update_vat_withhold_base_percent(env):
+    # For vat withhold taxes, replace factor_percent=12% with factor_percent=100%
```

Comment:
just a bit confusing that 12 -> 100 is in the tax repartition line, then why is the method named base_percent?

Reply:
Done, improved method name and comment